src/app/widgets/inlineinput/inlineinput.component.ts

```diff
@@ -139,6 +143,9 @@ export class InlineInputComponent implements OnInit {
           break;
       }
     }
+    if (this.isNotValid) {
+      this.errorMessage = `This is a ${this.type} field`;
+    }
   }
 
   checkInteger(keycode, event) {
```

Comment:
This function won't work. You're checking for each `keycode` individually. `checkInteger` returns true for string like `1.3.4.5.`.
A better approach to validating input in the `onSave` method by using `parseInt`/`parseFloat` methods.

Reply:
The motive was to show error when you type a character and don't see anything happening in the text field. Because we block keys to the respective fields.

Reply:
>Because we block keys to the respective fields.

We shouldn't do this. Let the user type whatever he wants to and then throw the error while saving.

Reply:
That's a UX discussion.

Reply:
The point is the component is capable of doing the both.